File: POOBanco/Python/Valor.py
from abc import ABC, abstractmethod

class Valor(ABC):
    __vale = 0.0
    __tipo = ""

    @abstractmethod
    def apresentacao(self):
        pass

    def __init__(self, vale, tipo):
        self.__vale = vale
        self.__tipo = tipo

    def __del__(self):
        print("Objeto Valor Destruído com Sucesso!")

    def getVale(self):
        return self.__vale

    def getTipo(self):
        return self.__tipo

    def setVale(self, vale):
        self.__vale = vale

    def setTipo(self, tipo):
        self.__tipo = tipo

File: POOBanco/PHP/Valor.php
<?php
abstract class Valor {
    protected $vale; // Float
    protected $tipo; // String

    public abstract function apresentacao();

    public function __construct($vale, $tipo) {
        $this->vale = $vale;
        $this->tipo = $tipo;
    }

    public function __destruct() {
        echo "<p>Objeto Valor Destruído com Sucesso!</p>";
    }

    public function getVale() {
        return $this->vale;
    }

    public function getTipo() {
        return $this->tipo;
    }

    public function setVale($vale) {
        $this->vale = $vale;
    }

    public function setTipo($tipo) {
        $this->tipo = $tipo;
    }
}